models/transport.go

```diff
@@ -201,5 +202,40 @@ func (transport *transport) FindEnqueuedDisk(diskUUID uuid.UUID) Disk {
 	return nil
 }
 
+func (transport *transport) DeleteVolume(userUUID uuid.UUID, volumeUUID uuid.UUID) (string, error) {
+	// TO DO: deletion process worker
+	var errCode string
+	var err error
+	var volume *Volume
+
+	// Retrieve volume from transport
+	volume = Transport.GetVolume(userUUID, volumeUUID)
+	if volume == nil {
+		return constants.TRANSPORT_VOLUME_NOT_FOUND, errors.New("Volume not found in transport layer")
+	}
+
+	// Remove enqueued files from volume
+	transport.FileUploadQueueMutex.Lock()
+	defer transport.FileUploadQueueMutex.Unlock()
+	for _, fc := range transport.FileUploadQueue {
+		if fc.File.GetVolume().UUID == volumeUUID {
+			delete(transport.FileUploadQueue, fc.File.GetUUID())
+		}
+	}
```

Comment:
to jest bardzo błędogenne, na razie zaimplementujmy usuwanie volume w ten sposób, że jak jest coś zaenqueuowane, to nie da się go usunąć - tak jak z dyskami. Jest taka funkcja transport.FindEnqueuedDisk, można zaimplementować podobną funkcję dla volumes. Jeśli jest coś zaenqueuowane na dany volume, to nie usuwajmy volume

Reply:
i chyba wolałbym, żeby ta walidacja była w controllerze

Reply:
👍